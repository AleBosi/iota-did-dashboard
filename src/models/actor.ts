// /models/actor.ts

export interface Actor {
  id: string; // DID unico (es: "did:iota:evm:0x...")
  name: string;
  role: "admin" | "azienda" | "creator" | "operatore" | "macchinario";
  aziendaId?: string;      // DID dell’azienda di appartenenza
  seed?: string;           // non esportare mai in VC!
  publicKey?: string;      // per interoperabilità/validazione
  createdAt?: string;      // opzionale, ISO date
  updatedAt?: string;      // opzionale

  // AGGIUNGI QUESTO CAMPO:
  vcIds?: string[];        // Lista degli ID delle VC associate a questo attore
}
