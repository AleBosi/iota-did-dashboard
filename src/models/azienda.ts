// /models/azienda.ts
export interface Azienda {
  id: string;         // DID (es: did:iota:evm:...)
  name: string;
  seed: string;       // per login e firma
  legalInfo?: {
    vat?: string;         // partita IVA o VAT
    lei?: string;         // Legal Entity Identifier (EBSI ready)
    address?: string;
    email?: string;
    country?: string;
    [key: string]: any;   // altri campi liberi
  };
  creators: string[];     // array di DID
  operatori: string[];
  macchinari: string[];
  createdAt?: string;     // ISO
  updatedAt?: string;     // ISO
}
