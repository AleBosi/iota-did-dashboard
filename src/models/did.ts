// /models/did.ts
export interface DidDocument {
  id: string; // es: "did:iota:evm:0x123..."
  controller?: string[];
  verificationMethod?: {
    id: string;
    type: string;
    publicKeyMultibase: string;
  }[];
  authentication?: string[];
  service?: {
    id: string;
    type: string;
    serviceEndpoint: string;
  }[];
  // Altri campi secondo specifica, se serve
}
