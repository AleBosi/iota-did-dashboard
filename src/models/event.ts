// /models/event.ts

export interface Event {
  id: string;
  productId: string;         // ID prodotto collegato (ora obbligatorio)
  operatoreId: string;       // ID operatore coinvolto
  macchinarioId: string;     // ID macchinario coinvolto
  type: string;              // es: "Assemblaggio", "Controllo", "Assegnazione"
  description: string;
  date: string;              // ISO
  creatorId: string;         // DID del creator che ha creato lâ€™evento
  done?: boolean;            // true se eseguito
  bomComponent?: string;     // opzionale, componente specifico
  proofId?: string;          // se firmerai evento come VC
  vcIds?: string[];          // VC associate/notarizzate collegate a questo evento
}
