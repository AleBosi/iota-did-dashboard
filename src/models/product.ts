// /models/product.ts
export interface Product {
  id: string;              // ID univoco locale/app
  did: string;             // DID prodotto (es: did:iota:evm:0x...)
  name: string;            // Nome commerciale o etichetta prodotto
  typeId?: string;         // riferimento a ProductType.id
  serial?: string;         // seriale prodotto
  owner?: string;          // DID del proprietario corrente
  bom?: { id: string; name: string }[]; // lista componenti (Bill of Materials)
  credentials?: string[];  // lista di VC associate (ID delle VC)
  description?: string;    // opzionale
}
