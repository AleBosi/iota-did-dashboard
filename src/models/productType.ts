// /models/productType.ts
export interface ProductType {
  id: string;               // es: GTIN, oppure UUID univoco
  name: string;
  description?: string;
}
