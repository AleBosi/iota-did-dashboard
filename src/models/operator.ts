// /models/operator.ts
import { Actor } from "./actor";

export interface Operator extends Actor {
  // aggiungi campi solo se servono SOLO agli operatori
}
