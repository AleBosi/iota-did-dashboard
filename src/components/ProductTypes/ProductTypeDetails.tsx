import { ProductType } from "../../models/productType";

interface Props {
  type: ProductType;
}

export default function ProductTypeDetails({ type }: Props) {
  if (!type) return null;
  return (
    <div className="border rounded p-4 bg-gray-50 mb-2">
      <div><b>Nome:</b> {type.name}</div>
      <div><b>ID:</b> {type.id}</div>
      {type.description && <div><b>Descrizione:</b> {type.description}</div>}
    </div>
  );
}
