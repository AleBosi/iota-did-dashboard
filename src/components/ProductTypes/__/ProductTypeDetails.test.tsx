import { render, screen } from "@testing-library/react";
import ProductTypeDetails from "../ProductTypeDetails";
import { ProductType } from "../../../models/productType";

describe("ProductTypeDetails", () => {
  const sampleType: ProductType = {
    id: "1234",
    name: "Tipo Test",
    description: "Descrizione test",
  };

  test("renderizza correttamente i dettagli", () => {
    render(<ProductTypeDetails type={sampleType} />);
    expect(screen.getByText(/nome:/i)).toBeInTheDocument();
    expect(screen.getByText(sampleType.name)).toBeInTheDocument();
    expect(screen.getByText(/id:/i)).toBeInTheDocument();
    expect(screen.getByText(sampleType.id)).toBeInTheDocument();
    expect(screen.getByText(/descrizione:/i)).toBeInTheDocument();
    expect(screen.getByText(sampleType.description!)).toBeInTheDocument();
  });

  test("non renderizza nulla se type Ã¨ null o undefined", () => {
    const { container } = render(<ProductTypeDetails type={null as any} />);
    expect(container).toBeEmptyDOMElement();
  });

  test("non mostra descrizione se non presente", () => {
    const typeNoDesc: ProductType = { id: "5678", name: "No Desc" };
    render(<ProductTypeDetails type={typeNoDesc} />);
    expect(screen.queryByText(/descrizione:/i)).toBeNull();
  });

  // Snapshot
  test("matcha lo snapshot con descrizione", () => {
    const { asFragment } = render(<ProductTypeDetails type={sampleType} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("matcha lo snapshot senza descrizione", () => {
    const typeNoDesc: ProductType = { id: "5678", name: "No Desc" };
    const { asFragment } = render(<ProductTypeDetails type={typeNoDesc} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
