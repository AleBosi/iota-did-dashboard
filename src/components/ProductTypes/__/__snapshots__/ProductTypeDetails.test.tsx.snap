// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductTypeDetails matcha lo snapshot con descrizione 1`] = `
<DocumentFragment>
  <div
    class="border rounded p-4 bg-gray-50 mb-2"
  >
    <div>
      <b>
        Nome:
      </b>
       Tipo Test
    </div>
    <div>
      <b>
        ID:
      </b>
       1234
    </div>
    <div>
      <b>
        Descrizione:
      </b>
       Descrizione test
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProductTypeDetails matcha lo snapshot senza descrizione 1`] = `
<DocumentFragment>
  <div
    class="border rounded p-4 bg-gray-50 mb-2"
  >
    <div>
      <b>
        Nome:
      </b>
       No Desc
    </div>
    <div>
      <b>
        ID:
      </b>
       5678
    </div>
  </div>
</DocumentFragment>
`;
